### PR TITLE
i-bem__dom: base `live` is not called in derived block (close #608)

### LINKED ISSUE #608: i-bem__dom: base `live` is not called in derived block
- `block1` is declared with `live` section
- `block1` processes its live section (DOM.init)
- `block2` is declared as derived from `block1` with `live` section contained base call within
- `block2` processes its live section

Unexpected result: base call within `live` section of `block2` doesn't work

### PR DESCRIPTION
**Pull request by dfilatov**
cc @veged @narqo

### CODE REVIEW

File: common.blocks/i-bem/__dom/i-bem__dom.spec.js [hunk 1]
```diff
@@ -811,6 +811,89 @@ describe('i-bem__dom', function() {
...
+        it('should properly use live after adding declaration', function() {
+            var spy1 = sinon.spy(),
+                spy2 = sinon.spy();
+
+            DOM.decl('block1', {}, { live : spy1 });
+
+            DOM.init(BEMHTML.apply({ block : 'block1', js : true }));
+
+            spy1.should.have.been.calledOnce;
+            spy2.should.not.have.been.called;
```
Comment (narqo) [inline common.blocks/i-bem/__dom/i-bem__dom.spec.js]: This assert looks useless, isn't it?

File: common.blocks/i-bem/__dom/i-bem__dom.js [hunk 2]
```diff
@@ -927,7 +927,15 @@ var DOM = BEM.decl('i-bem__dom',/** @lends BEMDOM.prototype */{
 
             if(noLive ^ heedLive) {
```
Comment (narqo) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: May be we could add some notes about `^` here? ;)

File: common.blocks/i-bem/__dom/i-bem__dom.spec.js (hunk 1 above)
Comment (dfilatov) [inline common.blocks/i-bem/__dom/i-bem__dom.spec.js]: Yes, you're definitely right. Removed.

File: common.blocks/i-bem/__dom/i-bem__dom.js (hunk 2 above)
Comment (dfilatov) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: I haven't changed this code )
Could you suggest reasonable comment?

Comment (narqo) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: Come on, it's your code, I don't understand it :P

Comment (dfilatov) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: it's equivalent to: `if((noLive && !heedLive) || (!noLive && heedLive)) {`

Comment (veged) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: `if(noLive ^ heedLive) { // should be always different from each other` ;-)

Comment (dfilatov) [inline common.blocks/i-bem/__dom/i-bem__dom.js]: added

**Comment by narqo** [comment]
Looks good

File: common.blocks/i-bem/__dom/i-bem__dom.spec.js [hunk 3]
```diff
@@ -811,6 +811,88 @@ describe('i-bem__dom', function() {
...
+                    spy2();
+                }
+            });
+
+            spy1.should.have.been.calledOnce;
+            spy2.should.have.been.calledOnce;
+
+            DOM.init(BEMHTML.apply({ block : 'block1', js : true }));
+
+            spy1.should.have.been.calledOnce;
```
Comment (veged) [inline common.blocks/i-bem/__dom/i-bem__dom.spec.js:889]: what the difference with `spy1.should.have.been.calledTwice` in pref test?

Comment (dfilatov) [inline common.blocks/i-bem/__dom/i-bem__dom.spec.js:889]: In previous spec two different blocks (`block1`, `block2` which derived from `block1`) are used.
In this spec `block1` is declared twice.

**Comment by veged** [comment]
:+1:

**Pull request merged**